### ISSUE TITLE
About the difference between yolo and coco dataset's num_class

### ISSUE THREAD

**Issue by JinRyua** [roboflow/rf-detr#624]
I saw the update of the yolo dataset support this time, and it seems that the number of num_classes learning varies depending on whether the same dataset is in yolo format or coco format.
Originally, I know that none class, the 0 index of coco, is learned together, I know it as N class + 1 (when learning)
It seems that this part is not included in the yolo format dataset, so it is learned by N class.
Is there a reason for the difference between these two, or am I not sure?

I think that if it's the same dataset, the learning results should be the same even if the format is different.

**Comment by Abdul-Mukit** [comment]
Interested in knowing as well. Both Roboflow and Ultralytics-YOLO datasets are 0-indexed for categories/classes. Only original-COCO-dataset (or from software like CVAT), annotations have 1-indexed classes. It doesn't make sense why the 
`num_classes = len(class_names) + 1` for roboflow data.

**Comment by isaacrob** [comment]
COCO should want 91, to do the +1 that you see in the Roboflow case. RF-DETR wants an extra logit to allow for assigning a sigmoid to background class during training (though not sure this implementation always uses it). The discrepancy is due to a bug that only causes an issue when using this repo to finetune on COCO which we don't do as we have a separate internal repo for pretraining. Fixing that is easiest via a refactor which we're going to be implementing soon thanks to the wonderful work by @Borda . Apologies for the confusion in the meantime.
